gstd/src/lock/mutex.rs

```diff
@@ -97,6 +103,8 @@ pub struct MutexLockFuture<'a, T> {
     mutex: &'a Mutex<T>,
 }
 
+/// In case of locked mutex and an `.await`, function `poll` checks if the mutex can be taken, else it waits.
```

Comment:
```suggestion
/// In case of locked mutex and an `.await`, function `poll` checks if the mutex can be taken, else it waits (goes into *waiting queue*).
```